### PR TITLE
fix ScaledValue comment on scaling

### LINKED ISSUE #50564: Sign error in comment on resource.ScaledValue
**Is this a BUG REPORT or FEATURE REQUEST?**:

> Uncomment only one, leave it on its own line: 
>
/kind bug
> /kind feature

**What happened**:
I invoked `ScaledValue(-9)` on a resource.Quantity, and got a result that was multiplied by a factor of 10^9.

**What you expected to happen**:
I expected to get a value that was multiplied by a factor of 10^-9, because the comment says "ScaledValue returns the value of ceil(q * 10^scale)".

**How to reproduce it (as minimally and precisely as possible)**:
```
q, _ := resource.ParseQuantity("10")
i := q.ScaledValue(-1)
```
print `i`.  You will find it is 100, not 1.

**Anything else we need to know?**:
Look at the meaning of the `AsScaledInt64` method invoked in the integer case.  This method returns the quantity expressed as a multiple of 10^scale.  10 is 100 tenths.

**Environment**:
- Kubernetes version (use `kubectl version`): master

### PR DESCRIPTION
**Pull request by dixudx**
**What this PR does / why we need it**:

**Which issue this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close that issue when PR gets merged)*: fixes #50564

**Special notes for your reviewer**:
/assign @MikeSpreitzer @wojtek-t @xiang90

**Release note**:

```release-note
None
```

### CODE REVIEW

**Comment by MikeSpreitzer** [comment]
/lgtm
But I think I am not yet officially a "member", so I am not sure what this is worth.

**Comment by MikeSpreitzer** [comment]
I can't request testing, but my `/ok-to-test` triggered testing.  Neat.

File: staging/src/k8s.io/apimachinery/pkg/api/resource/quantity.go [hunk 1]
```diff
@@ -705,7 +705,7 @@ func (q *Quantity) MilliValue() int64 {
 	return q.ScaledValue(Milli)
 }
 
-// ScaledValue returns the value of ceil(q * 10^scale); this could overflow an int64.
+// ScaledValue returns the value of ceil(q / 10^scale); this could overflow an int64.
```
Comment (wojtek-t) [inline staging/src/k8s.io/apimachinery/pkg/api/resource/quantity.go]: This isn't correct change. The comment is correct as it is now at head.

Comment (cheftako) [inline staging/src/k8s.io/apimachinery/pkg/api/resource/quantity.go]: As per https://github.com/kubernetes/kubernetes/blob/master/vendor/gopkg.in/inf.v0/dec.go#L41 the scale is the negated form of the exponent. That seems to line up with AsScaledInt64() which will call negativeScaleInt64 for a value with a scale of 0 and an explicit scale greater than 0. negativeScaleInt64 will then divide by 10 for each integer of scale.

**Comment by MikeSpreitzer** [comment]
I am sorry, but the comment is still wrong.  The comment still says "ScaledValue returns the value of ceil(q * 10^scale)".  Please look carefully at the code.  Following are some examples of evidence that the comment is wrong.

In the same file we see
```
// MilliValue returns the value of ceil(q * 1000); this could overflow an int64;
// if that's a concern, call Value() first to verify the number is small enough.
func (q *Quantity) MilliValue() int64 {
	return q.ScaledValue(Milli)
}
```
and in `amount.go` we see that `Milli Scale = -3`.  So `q.ScaledValue(-3)` returns the value of `ceil(q * 1000)` --- that is dividing by 10^-3, not multiplying.

Looking into the implementation of `ScaledValue`, we see that in one case it devolves to `q.i.AsScaledInt64(scale)`.  Looking over in `amount.go` at the comment on `AsScaledInt64`, we see the remark that "(1e20).AsScaledInt64(1) would result in overflow because 1e19 is not representable as an int64".  Note that 1e19 is 1e20 divided by 10^1, not multiplied.

**Pull request closed without merging**